chore: format java code

Christian-Schefe opened the pull request:
Code should be formatted for git diffs of my future commits to be readable. No idea why the code isn't formatted in the first place.
I haven't modified any code, I just ran the intellij idea formatter on all java files in server/src/main.

invaderb committed:
Merge pull request #10 from USTakAssociation/dev

Dev to Main

invaderb committed:
Merge pull request #11 from USTakAssociation/dev

Dev to Main

invaderb committed:
Merge pull request #12 from USTakAssociation/dev

Dev to Main

invaderb committed:
Merge pull request #14 from USTakAssociation/dev

Dev to Main | Update dependencies

invaderb committed:
Merge pull request #16 from USTakAssociation/dev

Dev to main

invaderb committed:
Merge pull request #20 from USTakAssociation/dev

Dev to Main

invaderb committed:
Merge pull request #22 from USTakAssociation/dev

Dev to Main

Christian-Schefe committed:
chore: format all java code

nitzel commented:
I think the main changes are converting tabs to spaces here.

nitzel commented on server/src/main/java/tak/Seek.java:
```diff
@@ -9,286 +9,282 @@
-	int rematchId;
+    public static enum COLOR {WHITE, BLACK, ANY}
+
+    ;
```
what kind of formatting is this where the semicolon gets pushed 2 lines down?

Ah, `public static enum NAME { Fields.. }` doesn't need a semicolon at all. I guess removing the semicolon here would make sense, though it doesn't change anything ^^

Christian-Schefe replied:
I saw that, but I wanted to clean up all the places where Intellij complains anyways in another pull request (after formatting is done, so one can actually review the diff). It currently shows me 245 warnings for the java code alone. This actually also includes most of the switches that can be turned into switch expressions instead. Also there are places where C-style arrays are used (String value[]) instead of (String[] value), I didn't even know that's possible.

Christian-Schefe commented:
> I think the main changes are converting tabs to spaces here.

That is most definitely a big part of the diffs, but there are also countless places with no proper spacing e.g. between operands: "a+b" instead of "a + b", and also the doc comments were expanded from one line into three. Also different line breaking at some places

nitzel reviewed (approved):
Looks good to me. A few places the formatting has made the readability slightly worse (e.g. switch), but I approve moving from tabs to spaces as well as some more uniformity for the usage of spaces in conditions and expressions.

# However
if we ever plan on moving forward with #7 then these changes will probably make merging changes a lot more annoying.

Christian-Schefe commented:
> if we ever plan on moving forward with #7 then these changes will probably make merging changes a lot more annoying.

How so? From what I can see, that PR is exclusively TypeScript code, which I haven't touched at all. I don't think there is even a single merge conflict between the two PRs

Christian-Schefe committed:
chore: revert to tabs, clean up imports and code style, add editorconfig

Christian-Schefe committed:
chore: adjust line break settings

Christian-Schefe commented:
I have now reverted to tabs as per bcreature's request. I have added a .editorconfig for consistent settings across editors (some settings are intellji-only, don't know if there are vscode equivalents, also vscode needs a plugin for .editorconfig iirc).
I optimized and reordered the imports and changed some minor code style issues.
The most notable ones are the c-style array syntax, unused exceptions in throw declarations, redundant final modifiers and that one comparison `0 >= sourceCount` turned into `sourceCount == 0` (as sourceCount is non-negative).

Christian-Schefe committed:
chore: adjust array initializer break settings

invaderb commented on server/src/main/java/tak/Game.java:
```diff
@@ -551,15 +536,14 @@ public GameDto toDto() {
-				.extraTimeAmount((int)(timeAmount / 1000))
+				.timeContingent((int) (originalTime / 1000))
+				.timeIncrement((int) (incrementTime / 1000))
+				.extraTimeAmount(timeAmount / 1000)
```
why'd you remove the (int) here but not on the 2 lines above?)

Christian-Schefe replied:
timeAmount is already an int, the cast is therefore redundant. The other two are longs that need to be cast.

invaderb reviewed (approved):
Looks good approved
